refinery/data_set_manager/views.py

```diff
@@ -497,29 +515,55 @@ def post(self, request, *args, **kwargs):
         if not request.is_ajax() or not request.body:
             return HttpResponseBadRequest()
 
-        file_data = json.loads(request.body)
         try:
-            base_path = file_data["base_path"]
+            file_data = json.loads(request.body)
+        except ValueError:
+            return HttpResponseBadRequest()
+        try:
+            input_file_list = file_data['list']
         except KeyError:
-            base_path = ""
+            return HttpResponseBadRequest()
+
+        try:
+            base_path = file_data['base_path']
+        except KeyError:
+            base_path = None
+        try:
+            identity_id = file_data['identityId']
+        except KeyError:
+            identity_id = None
 
         bad_file_list = []
         translate_file_source = generate_file_source_translator(
-            username=request.user.username, base_path=base_path)
-        # check if files are available
-        try:
-            for file_path in file_data["list"]:
-                # Explicitly check if file_path here is a string or unicode
-                # string
-                if not isinstance(file_path, unicode):
-                    bad_file_list.append(file_path)
-                else:
-                    file_path = translate_file_source(file_path)
-                    if not os.path.exists(file_path):
-                        bad_file_list.append(file_path)
-                logger.debug("Checked file path: '%s'", file_path)
-        except KeyError:  # if there's no list provided
-            return HttpResponseBadRequest()
+            username=request.user.username, base_path=base_path,
+            identity_id=identity_id
+        )
+
+        if settings.REFINERY_DEPLOYMENT_PLATFORM == 'aws':
+            # get a list of all uploaded S3 objects for the user
+            uploaded_s3_key_list = []
+            s3 = boto3.resource('s3')
+            s3_bucket = s3.Bucket(settings.MEDIA_BUCKET)
+            for s3_object in s3_bucket.objects.filter(
+                    Prefix='uploads/{}'.format(identity_id)
+            ):
+                uploaded_s3_key_list.append(s3_object.key)
+
+        for input_file_path in input_file_list:
+            if not isinstance(input_file_path, unicode):
```

Comment:
Actually, I'm simply re-using existing code here but I was curious about this too. May be @scottx611x could comment since he wrote it originally :)